### PR TITLE
feat!: rename rule shebang => hashbang, deprecate rule shebang

### PR DESCRIPTION
**Pull request by aladdin-add**
fixes https://github.com/eslint-community/eslint-plugin-n/issues/196

### CODE REVIEW

File: docs/rules/shebang.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
-# Require correct usage of shebang (`n/shebang`)
+# Require correct usage of hashbang (`n/shebang`)
 
-💼 This rule is enabled in the following [configs](https://github.com/eslint-community/eslint-plugin-n#-configs): ☑️ `flat/recommended`, 🟢 `flat/recommended-module`, ✅ `flat/recommended-script`, ☑️ `recommended`, 🟢 `recommended-module`, ✅ `recommended-script`.
+❌ This rule is deprecated. It was replaced by [`n/hashbang`](hashbang.md).
```
Comment (scagood) [inline docs/rules/shebang.md:3]: We should probably date the removal of shebang for like 6 months from now, or the next major version?

**Review by scagood — APPROVED** [review]
Good stuff :)

File: docs/rules/shebang.md (hunk 1 above)
Comment (aladdin-add) [reply docs/rules/shebang.md:3]: Not plan. It's fine to just keep the deprecated rules. let's just follow eslint's policy: https://eslint.org/docs/latest/use/rule-deprecation

Comment (scagood) [reply docs/rules/shebang.md:3]: That makes sense! :+1:

**Pull request merged**